internal/command/clusters/backup.go

```diff
@@ -0,0 +1,72 @@
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/jetstack/jsctl/internal/command/types"
+	"github.com/jetstack/jsctl/internal/kubernetes"
+	"github.com/jetstack/jsctl/internal/kubernetes/backup"
+)
+
+func Backup(run types.RunFunc, kubeConfigPath string) *cobra.Command {
+	var formatResources bool
+	var outputFormat string
+
+	var includeCertificates bool
+	var includeIssuers bool
+	var includeCertificateRequestPolicies bool
+
+	var skipIngressCerts bool
+
+	cmd := &cobra.Command{
+		Use:   "backup",
+		Short: "This command outputs the YAML data of Jetstack Secure relevant resources in the cluster",
+		Args:  cobra.MatchAll(cobra.ExactArgs(0)),
+		Run: run(func(ctx context.Context, args []string) error {
+			kubeCfg, err := kubernetes.NewConfig(kubeConfigPath)
+			if err != nil {
+				return err
+			}
+
+			opts := backup.ClusterBackupOptions{
+				RestConfig: kubeCfg,
+
+				FormatResources: formatResources,
+
+				IncludeCertificates:               includeCertificates,
+				IncludeIngressCertificates:        !skipIngressCerts,
+				IncludeIssuers:                    includeIssuers,
+				IncludeCertificateRequestPolicies: includeCertificateRequestPolicies,
+			}
+
+			clusterBackup, err := backup.FetchClusterBackup(context.Background(), opts)
+			if err != nil {
+				return fmt.Errorf("error backing up cluster: %s", err)
+			}
+
+			backupYAML, err := clusterBackup.ToYAML()
+			if err != nil {
+				return fmt.Errorf("error converting backup to YAML: %s", err)
+			}
+
+			fmt.Fprintf(os.Stdout, "%s", backupYAML)
+
+			return nil
+		}),
+	}
+
+	flags := cmd.PersistentFlags()
+	flags.BoolVar(&formatResources, "format-resources", true, "if set, will remove some fields from resources such as status and metadata to allow them to be cleanly applied later")
+	flags.StringVar(&outputFormat, "format", "yaml", "output format, one of: yaml, json")
+
+	flags.BoolVar(&includeCertificates, "include-certificates", true, "if set, certificate resources will be included in the backup")
+	flags.BoolVar(&includeIssuers, "include-issuers", true, "if set, issuer resources will be included in the backup")
+	flags.BoolVar(&includeCertificateRequestPolicies, "include-certificate-request-policies", true, "if set, certificate request policy resources will be included in the backup")
```

Comment:
As discussed, we have opted to eject in this case instead. afacf30b4106c43df819ebce522717c75aa94cf8 does this.

Reply:
I was wondering about this too. I thought that it needed to be v1, and have v1 with served: true (though I don't check that here).

My perhaps naive guess was that if v1 was not the first version, then the first version would be a later version, e.g. v2, and this would be served instead. Given this felt like a long way off and would create issuers for us, I thought we could just check the version of the first version item.

In summary, I think that if we have a v1 in the list, but it's not served, then we might have an issue?